src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -126,8 +136,30 @@ public void update(@RequestBody User user, @PathVariable long id)
         if (user.getId() != id) {
             throw new UserIdMismatchException();
         }
-        userRepository.findById(id)
-            .orElseThrow(UserNotFoundException::new);
+        User userFound = userRepository.findById(id)
+                             .orElseThrow(UserNotFoundException::new);
+        user.setPassword(userFound.getPassword());
+        userRepository.save(user);
+    }
+
+    /**
+     * Updates user password
+     *
+     * @param requestBody users password on body request
+     * @param id          identifications's user
+     * @throws UserNotFoundException
+     */
+    @PatchMapping("/{id}")
+    @ApiOperation(value = "Returns user updated")
+    @ApiResponses(value = {
+        @ApiResponse(code = 204, message = "Updates user correctly"),
+        @ApiResponse(code = 404, message = "User Not Found")
+    })
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void update(@RequestBody Map<String, Object> requestBody, @PathVariable long id)
```

Comment:
The update method should be different from the update password method, they should be two different endpoints. The endpoint that changes the password should receive the old password and the new password as parameters. It should verify that the old password parameter matches the current password and in that case, update the password to the new one 😄

Reply:
I user polymorphism so the method is duplicated but the parameters are different, I am going to change the name to more clarity and I do the other things too.